internal/controllers/provisioningrequest_clusterinstall.go

```diff
@@ -79,8 +81,16 @@ func (t *provisioningRequestReconcilerTask) renderClusterInstanceTemplate(
 				}
 
 				var disallowedChanges []string
-				if len(updatedFields) != 0 {
-					disallowedChanges = append(disallowedChanges, updatedFields...)
+
+				for _, updatedField := range updatedFields {
+					// Add "AgentClusterInstall" to ClusterInstance.SuppressedManifests in order to
+					// prevent unnecessary updates to ACI.
+					if updatedField == "clusterImageSetNameRef" &&
+						crProvisionedCond.Reason == string(utils.CRconditionReasons.Completed) {
+						suppressedManifests = append(suppressedManifests, "AgentClusterInstall")
```

Comment:
The template has to be usable in both install and upgrade cases. This is to prevent the provisioning failed state after updating the template ref to the new version.

Reply:
oh, okay, so for the upgrade case, clusterImageSetNameRef will need to be updated to the new version even if it's not actually used?